test/unit/test_str.c

```diff
@@ -3,6 +3,20 @@
 
 //TODO test r_str_chop_path
 
+bool test_r_str_format(void) {
+	r_strf_frame (4, 32);
```

Comment:
i really don't like this :( it is 1second that you initially have

r_str_frame (4), then someone needs another string and just use again r_strf but he does not increment r_str_frame and you end up with overwriting other strings pointers :/ it seems fragile to me.

Reply:
Also, it's easy that you then, after 100 commits, need another string (which may be longer than 32) and you end up with truncated strings.

Reply:
1 second?

it works exactly like sdb_fmt, the 4 is the amount of strings you will be using at the same time, its a cyclic array. well that's up to the developer, and we can add asserts to catch this

Reply:
and 99% of the usecases this is not necessary, and 99% of the situations it is pretty under control the kind of strings and sizes you will manage in a function, remember, this is local to the function, its not global like sdb_fmt